dm/docs/RFCS/20220124_enhance_task_manageability.md

```diff
@@ -0,0 +1,131 @@
+
+### Current State machine of task
+
+![dm old task lifecycle](../media/dm-old-task-lifecycle.png)
+
+### New State machine of task
+
+![dm new task lifecycle](../media/dm-new-task-lifecycle.png)
+
+This is mainly achieved by adding a new command `dmctl task create`, which creates a stopped task instead of creating and starting a new task with single command `dmctl start-task`.
+
+## Goals
+
+- Organize and optimize the state machine of tasks for better management
+- Unify style of commands in dmctl and optimize dmctl interaction experience
+- Add/modify OpenAPI, enhance eco-tools
+
+## Design and Examples
+
+New syntax of dmctl is `dmctl [resource type] [command] [flags] [object-name]`
+
+where `resource type`, `command` and `flags` are:
+
+- `resource type`: Specifies the resource you want to control. resource types are case-insensitive and there are limited resource types. currently there are only these types of resource types: `task`, `source`, `relay`, `ddl-lock` and `member`.
+
+- `command`: Specifies the operation that you want to perform on one or more resources, for example `create`, `get`, `update`, `delete`, `etc`.
+
+- `flags`: Specifies optional flags. For example, you can use the `--master-addr` flags to specify the address and port of the DM-Master server.
+
+- `object-name`: Specifies the name of the resource you want to operate, such as the name of the `task`.
+
+### dmctl commands for Task
+
+| command | Full Syntax Example                                                                                                              | Flags                                                             | Description                                                              |
+|---------|----------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------|--------------------------------------------------------------------------|
+| check   | `dmctl task check --config-file=dm.yaml --error-count=1 --warn-count=1 <task-name>`                                              | --config-file, --error-count, --warn-count                        | check the task config yaml file.                                         |
+| create  | `dmctl task create --config-file="dm.yaml"`                                                                                      | --config-file                                                     | create a stopped task with config file.                                  |
+| update  | `dmctl task update --config-file="dm.yaml <task-name>"`                                                                          | --config-file                                                     | update a stopped task with config file.                                  |
```

Comment:
```suggestion
| update  | `dmctl task update --config-file="dm.yaml" <task-name>`                                                                          | --config-file                                                     | update a stopped task with config file.                                  |
```

does this flags can be omitted? What's the behaviour then? (because I can see the syntax of this command and treat every flags optional by default)